test/hotspot/gtest/classfile/test_symbolTable.cpp

```diff
@@ -33,6 +33,8 @@ TEST_VM(SymbolTable, temp_new_symbol) {
   JavaThread* THREAD = JavaThread::current();
   // the thread should be in vm to use locks
   ThreadInVMfromNative ThreadInVMfromNative(THREAD);
+  // Disable the temp symbol cleanup delay queue because it increases refcounts.
+  TempNewSymbol::set_cleanup_delay_enabled(false);
```

Comment:
So we have additional check for "enabled" flag in hot production code only to make these tests happy? If so, can we "just" drain the delay queue after new_symbol here? Maybe with helper method here in test?

Reply:
Yes that's true. I can try to avoid it.

Reply:
I have added a small helper in the latest commit - do you think it's reasonable?